### ISSUE TITLE
Add helper methods NaN/Infinity for Complex type

### ISSUE THREAD

**Issue by kant2002** [dotnet/corefx#32225]
## Overview
When working with Complex numbers there now helper methods for dealing with NaN and Infinity values. For example with `double` it is possible check is value is `NaN` or `Infinity`

## API
This is an example of how the API could look like.

```csharp
public class Complex
{
    public static bool IsFinite(Complex value);
    public static bool IsNaN(Complex value);
    public static bool IsInfinity(Complex value);

    public static Complex  NaN { get; }
    public static Complex  Inifinity { get; }
}
```

/cc: @tannergooding

**Comment by tannergooding** [comment]
What is the proposed value for `Infinity` and `NaN`?

**Comment by kant2002** [comment]
I'm almost sure that only one sane way to return `NaN` is to return `(NaN, NaN)`. For `Infinity` is more likely personal opinion. I think that current implementation of `Complex` really model just single `Infinity` which happens to be represented in different ways. so I propose `(+Inf, +Inf)`

**Comment by tannergooding** [comment]
Resolved in #33184
